### PR TITLE
Adding public API test coverage for Aspire.MongoDB.Driver

### PR DESCRIPTION
**Pull request by Alirexaa**
Contributes to #5047 and #2649 

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5171)

### CODE REVIEW

File: tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+namespace Aspire.MongoDB.Driver.Tests;
+
+public class MongoDBClientPublicApiTests
+{
+    [Fact]
+    public void AddMongoDBClientShouldThrowWhenBuilderIsNull()
+    {
+        IHostApplicationBuilder builder = null!;
+
```
Comment (Zombach) [inline tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs:15]: I think this indentation is not needed

File: tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(builder), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddMongoDBClientShouldThrowWhenNameIsNull()
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
```
Comment (Zombach) [inline tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs:28]: And here

File: tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(connectionName), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddMongoDBClientShouldThrowWhenNameIsEmpty()
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
```
Comment (Zombach) [inline tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs:41]: And here

File: tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        var exception = Assert.Throws<ArgumentException>(action);
+        Assert.Equal(nameof(connectionName), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddKeyedMongoDBClientShouldThrowWhenBuilderIsNull()
+    {
+        IHostApplicationBuilder builder = null!;
+
```
Comment (Zombach) [inline tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs:54]: And here

File: tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(builder), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddKeyedMongoDBClientShouldThrowWhenNameIsNull()
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
```
Comment (Zombach) [inline tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs:67]: And here

File: tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(name), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddKeyedMongoDBClientShouldThrowWhenNameIsEmpty()
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
```
Comment (Zombach) [inline tests/Aspire.MongoDB.Driver.Tests/MongoDBDriverPublicApiTests.cs:80]: And here

**Review by sebastienros — APPROVED** [review]

**Pull request merged**